src/workerd/api/gpu/gpu-utils.c++

```diff
@@ -41,7 +41,38 @@ wgpu::FeatureName parseFeatureName(GPUFeatureName &str) {
     return wgpu::FeatureName::Float32Filterable;
   }
 
-  KJ_FAIL_REQUIRE("unknown GPU feature", str);
+  JSG_FAIL_REQUIRE(TypeError, "unknown GPU feature", str);
+}
+
+kj::Maybe<GPUFeatureName> getFeatureName(wgpu::FeatureName& feature) {
+  switch (feature) {
```

Comment:
You might consider implementing `KJ_STRINGIFY` for these enum structures. The implementation would be largely the same but it would allow you to do `kj::str(feature)`,  rather than `kj::str(getFeatureName(feature))`, etc. Search the code base for examples of where `KJ_STRINGIFY` has been implemented.

Reply:
I didn't use `KJ_STRINGIFY` because in this case I want to return an optional value. We may receive unknown features and I don't want to blow up but just ignore those features. `KJ_STRINGIFY` AFAICT, forces me to return some string.